source/drivers/MicroBitFile.cpp

```diff
@@ -0,0 +1,280 @@
+#include "MicroBitFile.h"
+#include "ErrorNo.h"
+
+/**
+  * Creates an instance of a MicroBitFile and creates a new file if required.
+  *
+  * @param fileName the name of the file to create/open.
+  *
+  * @param mode One of: READ, WRITE, READ_AND_WRITE. Defaults to READ_AND_WRITE.
+  */
+MicroBitFile::MicroBitFile(ManagedString fileName, int mode)
+{
+    this->fileName = fileName;
+
+    MicroBitFileSystem* fs;
+
+    if(MicroBitFileSystem::defaultFileSystem == NULL)
+        fs = new MicroBitFileSystem();
+    else
+        fs = MicroBitFileSystem::defaultFileSystem;
+
+    fileHandle = fs->open(fileName.toCharArray(), mode | CREATE);
```

Comment:
default open mode for MicroBitFile updated to READ | WRITE | CREATE as discussed, and an isValid() added.